packages/client/wallets/smart-wallet/src/types/Error.ts

```diff
@@ -1,57 +1,51 @@
-export class NotAuthorizedError extends Error {
-    code = "ERROR_NOT_AUTHORIZED";
-
-    constructor(message: string) {
+export const SmartWalletErrors = {
+    NOT_AUTHORIZED: "smart-wallet:not-authorized",
+    TRANSFER: "smart-wallet:transfer.error",
+    TRANSACTION: "smart-wallet:transaction.error",
+    CROSSMINT_SERVICE: "smart-wallet:crossmint-service.error",
+    RUNNING_ON_SERVER: "smart-wallet:running-on-server",
+    UNCATEGORIZED: "smart-wallet:uncategorized", // catch-all error code
+} as const;
+export type SmartWalletErrorCode = (typeof SmartWalletErrors)[keyof typeof SmartWalletErrors];
+
+export class SmartWalletSDKError extends Error {
+    public readonly code: SmartWalletErrorCode;
+
+    constructor(message: string, code: SmartWalletErrorCode = SmartWalletErrors.UNCATEGORIZED) {
         super(message);
-        Object.setPrototypeOf(this, NotAuthorizedError.prototype);
+        this.code = code;
     }
 }
 
-export class TransferError extends Error {
-    code = "ERROR_TRANSFER";
-
+export class NotAuthorizedError extends SmartWalletSDKError {
     constructor(message: string) {
-        super(message);
-        Object.setPrototypeOf(this, TransferError.prototype);
+        super(message, SmartWalletErrors.NOT_AUTHORIZED);
     }
 }
 
-export class TransactionError extends Error {
-    code = "ERROR_TRANSACTION";
+export class TransferError extends SmartWalletSDKError {
+    constructor(message: string) {
+        super(message, SmartWalletErrors.TRANSFER);
+    }
+}
 
+export class TransactionError extends SmartWalletSDKError {
     constructor(message: string) {
-        super(message);
-        Object.setPrototypeOf(this, TransactionError.prototype);
+        super(message, SmartWalletErrors.TRANSACTION);
     }
 }
 
-export class CrossmintServiceError extends Error {
-    public code = "ERROR_CROSSMINT_SERVICE";
+export class CrossmintServiceError extends SmartWalletSDKError {
     public status?: number;
 
     constructor(message: string, status?: number) {
-        super(message);
+        super(message, SmartWalletErrors.CROSSMINT_SERVICE);
         this.status = status;
-        Object.setPrototypeOf(this, CrossmintServiceError.prototype);
     }
 }
 
-export class RunningOnServerError extends Error {
-    public readonly code = "ERROR_RUNNING_ON_SERVER";
-
+export class RunningOnServerError extends SmartWalletSDKError {
     constructor() {
-        super("Smart Wallet SDK should only be used client side.");
-    }
-}
-
-/**
- * Generic undefined error
- */
-export class WalletSdkError extends Error {
-    code = "ERROR_UNDEFINED";
-
-    constructor(message: string) {
-        super(message);
-        Object.setPrototypeOf(this, WalletSdkError.prototype);
+        super("Smart Wallet SDK should only be used client side.", SmartWalletErrors.RUNNING_ON_SERVER);
```

Comment:
I'm not sure we need an error code for this scenario. To em error codes are useful for errors you must handle (e.g. not authorized maybe, or network errors or whatever), but this one is not an error you must handle - its one you'll get during development once 

So I think the error message is sufficient